patches/minecraft/net/minecraft/entity/passive/HorseArmorType.java.patch

```diff
@@ -0,0 +1,64 @@
+--- ../src-base/minecraft/net/minecraft/entity/passive/HorseArmorType.java
++++ ../src-work/minecraft/net/minecraft/entity/passive/HorseArmorType.java
+@@ -37,7 +37,6 @@
+         return this.ordinal();
+     }
+ 
+-    @SideOnly(Side.CLIENT)
+     public String func_188573_b()
+     {
+         return this.field_188587_f;
+@@ -55,6 +54,7 @@
+         return this.field_188586_e;
+     }
+ 
++    @Deprecated /**Forge: Use getByName. Ordinals of mod-added enum constants are dependent on load order, unlike names.**/
+     public static HorseArmorType func_188575_a(int p_188575_0_)
+     {
+         return values()[p_188575_0_];
+@@ -62,9 +62,14 @@
+ 
+     public static HorseArmorType func_188580_a(ItemStack p_188580_0_)
+     {
+-        return p_188580_0_.func_190926_b() ? NONE : func_188576_a(p_188580_0_.func_77973_b());
++        if (p_188580_0_.func_77973_b() instanceof net.minecraftforge.common.IHorseArmor)
++        {
++            return ((net.minecraftforge.common.IHorseArmor) p_188580_0_.func_77973_b()).getHorseArmorType(p_188580_0_);
++        }
++        return p_188580_0_.func_77973_b() == Items.field_151138_bX ? net.minecraft.entity.passive.HorseArmorType.IRON : p_188580_0_.func_77973_b() == Items.field_151136_bY ? net.minecraft.entity.passive.HorseArmorType.GOLD : p_188580_0_.func_77973_b() == Items.field_151125_bZ ? net.minecraft.entity.passive.HorseArmorType.DIAMOND : net.minecraft.entity.passive.HorseArmorType.NONE;
+     }
+ 
++    @Deprecated //Forge: Use getByItemStack
+     public static HorseArmorType func_188576_a(Item p_188576_0_)
+     {
+         if (p_188576_0_ == Items.field_151138_bX)
+@@ -81,8 +86,29 @@
+         }
+     }
+ 
++    @Deprecated //Forge: Use ItemStack sensitive overload
+     public static boolean func_188577_b(Item p_188577_0_)
+     {
+         return func_188576_a(p_188577_0_) != NONE;
+     }
++    
++    /********* FORGE *********/
++    //Allows for textures located outside the vanilla horse armor folder
++    private HorseArmorType(String textureLocation, int armorStrengthIn)
++    {
++        this.field_188588_g = armorStrengthIn;
++        this.field_188586_e = textureLocation;
++        this.field_188587_f = net.minecraftforge.client.ForgeHooksClient.generateHorseArmorHash();
```

Comment:
Although this is only used on the client, this should not be in `ForgeHooksClient` for its existence on the server side.